client/src/components/rooms/RoomsMeetingExperience.tsx

```diff
@@ -2,33 +2,228 @@
+import {
+  connectToCallAutomation,
+  fetchTranscriptionStatus,
+  getCallSummaryFromServer,
+  LocaleCode,
+  startTranscription,
+  stopTranscription,
+  SummarizeResult
+} from '../../utils/CallAutomationUtils';
+import { Call, TeamsCall } from '@azure/communication-calling';
+import { SlideTextEdit20Regular } from '@fluentui/react-icons';
+import { TranscriptionOptionsModal } from './transcriptionNotifications/TranscriptionOptionsModal';
+import { CustomNotifications } from './transcriptionNotifications/CustomNotifications';
 
 export interface RoomsMeetingExperienceProps {
   roomsInfo: RoomsInfo;
   token: string;
   postCall?: PostCallConfig;
   fluentTheme?: PartialTheme | Theme;
   onDisplayError(error: any): void;
+  // this needs to be re-thought. we need a way for the config to say we are not using transcription, but also
+  // auto start, and transcription only
+  transcriptionClientOptions?: TranscriptionClientOptions;
 }
 
 const RoomsMeetingExperience = (props: RoomsMeetingExperienceProps): JSX.Element => {
-  const { roomsInfo, token, postCall, fluentTheme, onDisplayError } = props;
+  const { roomsInfo, token, postCall, fluentTheme, onDisplayError, transcriptionClientOptions } = props;
   const { userId, userRole, locator } = roomsInfo;
+  const [transcriptionStarted, setTranscriptionStarted] = useState(false);
+  const [showTranscriptionModal, setShowTranscriptionModal] = useState(false);
+  const [summarizationLanguage, setSummarizationLanguage] = useState<LocaleCode>('en-US');
+  const [summary, setSummary] = useState<SummarizeResult>();
+  const [summarizationStatus, setSummarizationStatus] = useState<'None' | 'InProgress' | 'Complete'>('None');
+  const [statefulClient, setStatefulClient] = useState<StatefulCallClient>();
+  const [callAgent, setCallAgent] = useState<DeclarativeCallAgent>();
+  const [call, setCall] = useState<Call | TeamsCall | undefined>();
+  const [callAdapter, setCallAdapter] = useState<CommonCallAdapter>();
+  const [customNotications, setCustomNotifications] = useState<ActiveNotification[]>([]);
+  const [callConnected, setCallConnected] = useState(false);
+  const [serverCallId, setServerCallId] = useState<string | undefined>(undefined);
+  const [transcriptionStartedByYou, setTranscriptionStartedByYou] = useState(false);
+
+  const transcriptionFeatureEnabled = transcriptionClientOptions?.transcription !== 'none';
+  const summarizationFeatureEnabled = transcriptionClientOptions?.summarization;
+  const theme = useTheme();
+  const callAutomationStarted = useRef(false);
+  const eventSourceRef = useRef<EventSource | null>(null);
+
+  useEffect(() => {
+    let eventSource: EventSource | null = null;
+
+    if (serverCallId && transcriptionFeatureEnabled) {
+      // Create EventSource connection when serverCallId is available. The URL provided here is for your server.
+      eventSource = new EventSource(`http://localhost:8080/api/notificationEvents`);
```

Comment:
The comment here doesn't feel distinct enough to warrant attention. 
Can we better indicate this may need to be changed to their URL?

Reply:
Agreed, does this break when deployed? We often use an environment variable for raw urls, or if its relative can it just be 
```suggestion
      eventSource = new EventSource(`/api/notificationEvents`);
```
^would something like that work with event source?

Reply:
this does break when deployed but this is retooled later on to get this from the app configuration which fetches it from the server env variables or the `defaultConfig.json` file depending on how contoso is setting up their environment